dll/win32/shell32/CDefView.cpp

```diff
@@ -612,6 +612,33 @@ void CDefView::UpdateStatusbar()
     }
 }
 
+unsigned __stdcall CDefView::_UpdateStatusbarProc(void *args)
+{
+    CDefView* pView = static_cast<CDefView*>(args);
+    pView->AddRef();
+    pView->UpdateStatusbarWorker(pView->m_hUpdateStatusbarThread);
+    pView->Release();
+    return 0;
+}
+
+void CDefView::UpdateStatusbar()
+{
+    HANDLE hOldThread = m_hUpdateStatusbarThread;
+
+    // We have to initialize m_hUpdateStatusbarThread before the target thread begins.
+    // So, we use CREATE_SUSPENDED.
+    HANDLE hNewThread = reinterpret_cast<HANDLE>(_beginthreadex(NULL, 0, _UpdateStatusbarProc,
```

Comment:
Ideally the `AddRef` should be before this call

Reply:
OK. Thank you.

Reply:
The purpose is to ensure that the object cannot be destroyed while the thread is starting/running.